### ISSUE TITLE
Resolving and verifying MConnection receiving buffer size constraints (for enhanced handling of transactions in large blocks)

### ISSUE THREAD

**Issue by staheri14** [celestiaorg/celestia-core#1190]
## Problem
In the course of working on the performance evaluation of `MConnection` #1162, I came across a potential issue concerning the usage of `MConnection`, particularly when it comes to increasing block sizes beyond current standards, for example, to 1 GB. The summary of the problem is that large messages, such as txs, can be dropped from circulation within the network due to the size limitations of the channel IDs' receiving buffer.

This issue is primarily rooted in the size limitations of the `MConnection`'s channel IDs' [receiving buffer](https://github.com/celestiaorg/celestia-core/blob/34b83b51889c92a9818221bb01dc78a35dc55c77/p2p/conn/connection.go#L751). Each channel ID has a predefined maximum message size, determined by the channel's [`RecvMessageCapacity`](https://github.com/celestiaorg/celestia-core/blob/34b83b51889c92a9818221bb01dc78a35dc55c77/p2p/conn/connection.go#L726) configuration. Currently, this limit is set to [21 MB](https://github.com/celestiaorg/celestia-core/blob/34b83b51889c92a9818221bb01dc78a35dc55c77/p2p/conn/connection.go#L42) by default, which means no message can exceed this size. As a result, any message larger than 21 MB will consistently fail to be received. This default value does not seem to be modified or checked on the celestia-app side (as far as I checked).

For instance, the size of a transmitted transaction could potentially be as large as the entire data square, as detailed in the formula found [here](https://github.com/celestiaorg/celestia-app/blob/454db16edb4e168d62105e86f953a81eea48c400/app/default_overrides.go#L244). In the case of a 1GB block, this implies that the message size could be 1GB. Therefore, it's crucial to ensure that the receiving buffer limit of the channel ID is set high enough to accommodate such large potential messages. If not, these large messages or transactions will likely be [dropped](https://github.com/celestiaorg/celestia-core/blob/34b83b51889c92a9818221bb01dc78a35dc55c77/p2p/conn/connection.go#L861) from circulation within the network due to buffer limitations.

## Acceptance Criteria 
- [ ] I plan to develop tests to verify this issue on the celestia-core side and then
- [ ]  implement a suitable fix in the celestia-app to address this limitation.

**Comment by evan-forbes** [comment]
This is an important point that we have to keep in the back of our heads. If I'm understanding this correctly, assuming that all of our gossiping mechanisms chunk data, we should never hit this value unless a single tx or block part is larger than 21MB, correct? 
 
For example, we are able to create 100MB blocks in testground because we break it into 64KB block parts. I could be missing something, but I think the only time that this isn't done is during block sync?

**Comment by staheri14** [comment]
> we should never hit this value unless a single tx or block part is larger than 21MB, correct?

Right, while the maximum size for data transmitted to a channel should be under 21MB, we alternatively have the option to configure channel IDs (through their descriptors) to accommodate larger messages. It's crucial to either properly set this size limit or compare the size of messages against this maximum. Failing to do so results in messages being discarded (even worse, the connection gets dropped).

> For example, we are able to create 100MB blocks in testground because we break it into 64KB block parts. I could be missing something, but I think the only time that this isn't done is during block sync?

That is correct, since the individual block parts are smaller than 21MB, no problem occurs on that side.

**Comment by staheri14** [comment]
I’ve addressed the first action item in this issue, but the second one is still pending a decision. Deferring to @evan-forbes for how to proceed.
